xbmc/epg/EpgContainer.cpp

```diff
@@ -245,16 +245,17 @@ bool CEpgContainer::PersistAll(void)
 {
   bool bReturn(true);
   m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
+  EPGMAP copy = m_epgs;
   m_critSection.unlock();
 
-  for (EPGMAP_CITR it = copy.begin(); it != copy.end() && !m_bStop; it++)
+  for (const auto &epgEntry : copy)
   {
-    CEpg *epg = it->second;
+    CEpg *epg = epgEntry.second;
     if (epg && epg->NeedsSave())
-    {
       bReturn &= epg->Persist();
-    }
+
+    if (m_bStop)
```

Comment:
This is not my code at all so feel free to ignore this comment but it took me a few minutes to figure out that the new code doesn't change behaviour (I actually already wrote a comment before I realized that it wasn't the case) because it is extremely easy to miss the `&& !m_bStop` in the for loop definition. The new code is much more obvious to read for me.